core/ecschema-metadata/src/Interfaces.ts

```diff
@@ -94,3 +96,16 @@ export interface HasMixins {
   mixins: LazyLoadedMixin[];
   getMixinsSync(): Iterable<Mixin>;
 }
+
+/** This interface is implemented by a class that would provide formats for use in formatting quantities.
+ * @beta
+ */
+export interface FormatsProvider {
+  getFormat(name: string, unitSystem?: string): Promise<SchemaItemFormatProps | undefined>;
+  addFormat(name: string, formatProps: SchemaItemFormatProps, unitSystem?: string): Promise<void>;
+  removeFormat(name: string): Promise<void>;
+
+  onFormatUpdated: BeUiEvent<string>;
+  onFormatRemoved: BeUiEvent<string>;
+  onCacheCleared: BeUiEvent<void>;
+}
```

Comment:
Should a "provider" type of interface define "add", "remove" methods and events like "on removed" and "on cache cleared"? It looks like these members are implementation detail. Can't a provider be read-only without ability to add / remove (in fact, it seems `SchemaFormatsProvider` should be read-only)? What if provider doesn't have a cache?

IMO, it's enough to have a "get" method and one of these:
- `onFormatChanged` event for consumers to know that specific format has changed,
- `onFormatsChanged` event for consumers to know that any of the provided formats changed.

Reply:
this makes sense, I've modified the events naming and dropped the add/remove methods from the generic interface.

Reply:
I was suggesting to add one of the two events. Why do we need both? Especially since they're not raised consistently (`onFormatChanged` is raised on add/remove and `onFormatsChanged` is called on clear).

Reply:
Oops, I've removed `onFormatChanged`, keeping `onFormatsChanged` instead.